util/test/metric_test.cpp

```diff
@@ -69,4 +71,32 @@ TEST(MetricsTest, Aggregator) {
   ASSERT_EQ(1, aggregator->GetCounter(c.Name(), "messages_sent").Get());
 }
 
+// ToJson is a simple hand written serializer. We don't have a corresponding
+// deserializer, since it isn't strictly necessary. We use the builtin json_pp
+// program included with Ubuntu to verify that the output is valid json.
+//
+// System tests will actually use the JSON output, so we'll get extra validation
+// there.
+TEST(MetricTest, ToJson) {
+  auto aggregator = std::make_shared<Aggregator>();
+  Component c("replica", aggregator);
+  c.RegisterGauge("connected_peers", 3);
+  c.RegisterGauge("total_peers", 4);
+  c.RegisterStatus("state", "primary");
+  c.RegisterStatus("commit_path", "SLOW");
+  c.RegisterCounter("messages_sent", 0);
+  c.RegisterCounter("messages_received", 1);
+  c.Register();
+
+  Component c2("state-transfer", aggregator);
+  c2.RegisterGauge("blocks-remaining", 4);
+  c2.RegisterStatus("state", "sending-blocks");
+  c2.Register();
+
+  ostringstream oss;
+  oss << "echo '" << aggregator->ToJson() << "' | json_pp" << endl;
+
+  ASSERT_EQ(0, system(oss.str().c_str()));
```

Comment:
It took me a second to understand what was being tested here. Maybe add a tiny comment right above this, noting that json_pp will exit non-zero if it can't parse the input?

Reply:
As Bryan and I discussed offline, I'm going to change this test to eval python3 so that we don't rely on an extra program like json_pp.